R/arrays.R

```diff
@@ -50,6 +50,8 @@ array_branch <- function(array, margin = NULL) {
 
   if (length(margin) == 0) {
     list(array)
+  } else if (identical(margin, 1) && is.null(dim(array))) {
```

Comment:
The failure this is fixing was from an example, could you capture it in the unit tests as well please?

Reply:
Will do!

Reply:
Do you think the branch should be `if (is.null(dim(array)))`, and then we'd check for the margin and give an informative error message if it's not `1`?

By the way, `identical(1, 1L)` is `FALSE`. I think we need `is_integerish(margin, n = 1) && margin == 1`. Could you capture the integer case in the tests as well please?